libcoap/src/session/mod.rs

```diff
@@ -443,7 +443,9 @@ impl<'a> CoapSession<'a> {
     /// Panics if the given pointer is a null pointer.
     ///
     /// # Safety
-    /// The provided pointer must be valid.
+    /// The provided pointer must be valid, the provided session's app data must be a valid argument
+    /// to `CoapFfiRawCell<CoapClientSessionInner>::clone_raw_rc` or
+    /// `CoapFfiRawCell<CoapServerSessionInner>::clone_raw_rc` (depending on the session type).
```

Comment:
Are there square brackets missing?

```suggestion
    /// to [`CoapFfiRawCell<CoapClientSessionInner>::clone_raw_rc`] or
    /// [`CoapFfiRawCell<CoapServerSessionInner>::clone_raw_rc`] (depending on the session type).
```